coderd/prometheusmetrics/aggregator.go

```diff
@@ -68,7 +67,32 @@ type annotatedMetric struct {
 var _ prometheus.Collector = new(MetricsAggregator)
 
 func (am *annotatedMetric) is(req updateRequest, m *agentproto.Stats_Metric) bool {
-	return am.username == req.username && am.workspaceName == req.workspaceName && am.agentName == req.agentName && am.Name == m.Name && slices.Equal(am.Labels, m.Labels)
+	return am.username == req.username &&
+		am.workspaceName == req.workspaceName &&
+		am.agentName == req.agentName &&
+		am.templateName == req.templateName &&
+		am.Name == m.Name &&
+		labelsEqual(am.Labels, m.Labels)
+}
+
+func labelsEqual(a, b []*agentproto.Stats_Metric_Label) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	am := make(map[string]string)
+	for _, l := range a {
```

Comment:
nit: this is personal, but I used to misread `l`, `I`, `1`, and `|`. Maybe `lbl`?

Reply:
Sounds like time to pick a different font 😝 !